### PR TITLE
Export used errors in the AST.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by ekpyron — COMMENTED** [review]
Maybe a test case?

**Comment by ekpyron** [comment]
Also seems like `toJson` needs another overload or be generalized to all kinds of pointers in vectors.

**Comment by chriseth** [comment]
Yes, this PR is mainly a reminder that e still need that for the release. Please take over if you want.

**Comment by hrkrshnn** [comment]
What is left to be done here? Probably a changelog entry?

**Comment by chriseth** [comment]
Added changelog entry.

File: test/libsolidity/ASTJSON/used_errors_parseOnly.json [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            "src": "97:0:1"
+          },
+          "src": "72:33:1",
+          "stateMutability": "pure",
+          "virtual": false,
+          "visibility": "public"
+        }
+      ],
+      "src": "40:67:1",
+      "usedErrors": []
```
Comment (hrkrshnn) [inline test/libsolidity/ASTJSON/used_errors_parseOnly.json:168]: Should parse only include the `error T()`?

Comment (chriseth) [reply test/libsolidity/ASTJSON/used_errors_parseOnly.json:168]: No, because we need to do inheritance analisys

Comment (chriseth) [reply test/libsolidity/ASTJSON/used_errors_parseOnly.json:168]: or at least reference resolution.

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**